content/v2.0/architecture/overview.md

```diff
@@ -0,0 +1,206 @@
+---
+title: "Overview"
+order: 10
+---
+
+Hanami is based on two principles: 
+- Clean Architecture
+- [Monolith First](http://martinfowler.com/bliki/MonolithFirst.html).
+
+And the reason for this is to encourage** building applications easily and efficiently at the beginning**, but at the same time, that our **projects scale well when they grow**.
+
+## Clean Architecture
+
+The main purpose of this architecture is to enforce a **separation of concerns** between the **core** of our product and the **delivery mechanisms**.
+The first is expressed by the set of **use cases** that our product implements, while the latter are interfaces to make these features available to the outside world.
+
+When we generate a new project we can find two important directories: `lib/` and `slices/`.
+They are home to the main parts described above.
+
+### Application Core
+
+We implement a set of functionalities, without worrying about how they can be exposed to the outside world.
+This is the **cornerstone** of our product, and we want to be careful on how we manage dependencies for it.
+
+Within the `/lib` folder, you can place the code common for the whole application, like base classes and core logic.
+
+> `Hanami::Model` is the default choice for persisting our Ruby objects.
+This is a _soft-dependency_, it can be removed from our `Gemfile` and replaced with something else.
+
+Let's have a look at how the `lib/` directory appears for a new generated project called `bookshelf` that uses `Hanami::Model`.
+
+```shell
+lib
+├── bookshelf
+│   ├── action.rb
+│   ├── entities
```

Comment:
@swilgosz No biggie, but maybe a tailing slash for directories would improve the readability, e.g.

```
├── entities/
├── entities.rb
├── foodir/
    └── barfile
```

Reply:
good idea! BTW: not sure, if you are aware, but you can suggest changes inside reviews, that I can accept directly as your commit

Reply:
@swilgosz IC, will do next time.